From 92d8cc6ce85de08ffa073aba2463171b8dbf4b77 Mon Sep 17 00:00:00 2001
From: Adam Duskett <Aduskett@gmail.com>
Date: Thu, 28 Oct 2021 13:34:16 -0700
Subject: [PATCH] revert commit b13d3f3f724bf0ce44200760c81fbbccc07f6302

Commit b13d3f3f724bf0ce44200760c81fbbccc07f6302 forces all dependent libraries
to link statically, which is undersirable when building in a cross-compiled
environment that may not be compiling static libraries. Especially since many
of those libraries either compile shared libraries OR static libraries, but not
both at the same time.

Signed-off-by: Adam Duskett <Aduskett@gmail.com>
---
 setup.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/setup.py b/setup.py
index eef396b..ec56b26 100644
--- a/setup.py
+++ b/setup.py
@@ -253,8 +253,6 @@ def awscrt_ext():
 
     else:  # unix
         # linker will prefer shared libraries over static if it can find both.
-        # force linker to choose static variant by using using "-l:libcrypto.a" syntax instead of just "-lcrypto".
-        libraries = [':lib{}.a'.format(x) for x in libraries]
         libraries += ['rt']
 
     if distutils.ccompiler.get_default_compiler() != 'msvc':
-- 
2.32.0

